### PR TITLE
Added cpp dir and tooling

### PR DESCRIPTION
**Pull request by stevenewald**

### CODE REVIEW

File: src/cppsrc/test/src/desk_monorepo_test.cpp [hunk 1]
```diff
@@ -0,0 +1,8 @@
+#include <catch2/catch_test_macros.hpp>
```
Comment (echavemann) [inline src/cppsrc/test/src/desk_monorepo_test.cpp:1]: can we use gtest?

Comment (echavemann) [reply src/cppsrc/test/src/desk_monorepo_test.cpp:1]: probably best we have new stuff align with intern project so the knowledge carries

**Comment by stevenewald** [comment]
#47

File: .github/workflows/cpp-ci.yml [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+on:
+  push:
+    branches:
+      - main
+
+  pull_request:
+    branches:
+      - main
+
+jobs:
```
Comment (echavemann) [inline .github/workflows/cpp-ci.yml:12]: actions v4 and python setup v5 i believe, for nodejs warnings

File: .github/workflows/cpp-ci.yml [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+
+    - name: Install LLVM 18
+      run: |
+          wget -qO- https://apt.llvm.org/llvm-snapshot.gpg.key | sudo tee /etc/apt/trusted.gpg.d/apt.llvm.org.asc
+          sudo apt-add-repository "deb http://apt.llvm.org/jammy/ llvm-toolchain-jammy-18 main"
+          sudo apt update
+          sudo apt install llvm-18 llvm-18-dev llvm-18-tools clang-18 clang-tidy-18 clang-format-18 clang-tools-18 libclang-18-dev -y
+
+    - name: Install codespell
+      run: pip3 install codespell
```
Comment (echavemann) [inline .github/workflows/cpp-ci.yml:32]: pipx? (nino?)

File: .github/workflows/cpp-ci.yml (hunk 2 above)
Comment (echavemann) [reply .github/workflows/cpp-ci.yml:12]: also something to be said for seperate files like in the python ones, but im more or less fine with this one

File: .github/workflows/cpp-ci.yml [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+
+    steps:
+    - uses: actions/checkout@v3
+      with:
+        submodules: 'recursive'
+
+    - uses: actions/setup-python@v4
+      with: { python-version: "3.12" }
+
+    - name: Install LLVM 18
```
Comment (echavemann) [inline .github/workflows/cpp-ci.yml:24]: is there not a good action for this? if not im fine with it just surprised

File: .github/workflows/cpp-ci.yml [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+      id: cache-conan
+      uses: actions/cache@v3
+      env:
+        cache-name: cache-conan-deps
+      with:
+        path: ~/.conan2
+        key: ${{ runner.os }}-builder-${{ env.cache-name }}-${{ hashFiles('conanfile.py') }}
+        restore-keys: ${{ runner.os }}-builder-${{ env.cache-name }}-
+
+    - name: Install dependencies
```
Comment (echavemann) [inline .github/workflows/cpp-ci.yml:95]: i dont think we need numpy? Pipx install conan? unsure about libssl

File: .github/workflows/cpp-ci.yml [hunk 6]
```diff
@@ -0,0 +1,123 @@
...
+      shell: bash
+      run: |
+        pip3 install conan numpy
+        
+        sudo apt install libssl-dev -y
+        
+        bash < .github/scripts/conan-profile.sh
+        conan install . -s build_type=Release -b missing
+
+    - name: Setup MultiToolTask
```
Comment (echavemann) [inline .github/workflows/cpp-ci.yml:105]: do we need this?

File: CMakeLists.txt
Comment (echavemann) [inline CMakeLists.txt:1]: theres an argument for making this CMake less like NUTC and more like the simplified one for the intern project, but idrc either way

File: Makefile [hunk 7]
```diff
@@ -1,26 +1,40 @@
...
 
-unit:
+py-unit:
 	poetry run pytest src/pysrc/test/unit
 
-integration:
+py-integration:
 	poetry run pytest src/pysrc/test/integration
 
+cpp-deps:
```
Comment (echavemann) [inline Makefile:27]: nit: cpp-install? match with py?

File: Makefile [hunk 8]
```diff
@@ -1,26 +1,40 @@
...
+cpp-init:
+	cmake --preset=dev
+
+cpp-build:
+	cmake --build --preset=dev -j
+
+cpp-test:
+	ctest --preset=dev --timeout 1
+
+cpp-fmt:
```
Comment (echavemann) [inline Makefile:39]: same with format here

File: Makefile [hunk 9]
```diff
@@ -1,26 +1,40 @@
...
+cpp-deps:
+	conan install . -s build_type=Debug -b missing -pr cpp23 -pr:b cpp23
+
+cpp-init:
+	cmake --preset=dev
+
+cpp-build:
+	cmake --build --preset=dev -j
+
+cpp-test:
```
Comment (echavemann) [inline Makefile:36]: whats timeout here for

File: mypy.ini [hunk 10]
```diff
@@ -5,6 +5,7 @@ check_untyped_defs = True
 warn_return_any = True
 show_error_codes = True
 warn_unused_ignores = True
+exclude = conanfile.py
```
Comment (echavemann) [inline mypy.ini:8]: good catch lol

**Comment by echavemann** [comment]
I was envisioning more 
src/
- Cpp project one
- Pysrc
- Cpp project two
- etc

**Comment by stevenewald** [comment]
some of the unnecessary stuff was cuz i literally just copied this over, ill remove

**Comment by echavemann** [comment]
> #47

this doesnt link it to autoclose the issue, if its not in the desc you have to attach it yourself

**Comment by stevenewald** [comment]
@rewong03 too lazy to use act to test GitHub actions

**Comment by stevenewald** [comment]
> > #47
> 
> this doesnt link it to autoclose the issue, if its not in the desc you have to attach it yourself

im aware, just wanted to link it

**Pull request closed without merging**